File: ui/src/types/secret.ts
import type { ObjectMeta } from "./shared";

export interface Secret {
  metadata: ObjectMeta;
  type?: string;
  data?: Record<string, string>;
  immutable?: boolean;
}

export interface SecretList {
  items: Secret[];
}

export interface CreateSecretRequest {
  name: string;
  namespace: string;
  labels?: Record<string, string>;
  annotations?: Record<string, string>;
  type?: string;
  stringData?: Record<string, string>;
  immutable?: boolean;
}

File: backend/internal/builders/secret.go
package builders

import (
	"github.com/podforge/backend/internal/types"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func BuildSecret(req types.CreateSecretRequest) *corev1.Secret {
	secretType := corev1.SecretType(req.Type)
	if secretType == "" {
		secretType = corev1.SecretTypeOpaque
	}

	return &corev1.Secret{
		ObjectMeta: metav1.ObjectMeta{
			Name:        req.Name,
			Namespace:   req.Namespace,
			Labels:      req.Labels,
			Annotations: req.Annotations,
		},
		Type:       secretType,
		StringData: req.StringData,
		Data:       req.Data,
		Immutable:  req.Immutable,
	}
}
